### PR TITLE
refactor!: Remove now-obsolete property `skipZMiddleBinSearch`

### PR DESCRIPTION
**Pull request by CarloVarni**
This needs https://github.com/acts-project/acts/pull/2718 to be in.

As quoted there: The grid iterator will be able to iterate on whatever sequence of bins you like.

In other words... for a usual seed finding process we have:

`zBinsCustomLooping = [1, 2, 3, 4, 11, 10, 9, 8, 6, 5, 7]`
but for fast tracking you can simply provide a shorter vector. For instance:

`zBinsCustomLooping = [11, 10, 9, 8, 6, 5, 7]`
much simpler imho.

### CODE REVIEW

**Review by LuisFelipeCoelho — DISMISSED** [review]

**Comment by paulgessinger** [comment]
Is this ready to review @CarloVarni?

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [ee1e504a5f1d44a3044459e2254bc36e8caea4c6]
### :white_check_mark: All tests successful

|status|job|report|
|------|---|------|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790770)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790769)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790768)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790767)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790766)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790765)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790764)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790763)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790762)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790761)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790760)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790759)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790758)||
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790757)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790756)||
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790755)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790754)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790753)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790752)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790751)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790750)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790749) | [:paperclip:](https://acts-athena-outputs.web.cern.ch/test_data18_13TeV_1000evt/pulls/2835/)|
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34790748) | [:paperclip:](https://acts-athena-outputs.web.cern.ch/test_ttbarPU40_reco/pulls/2835/)|

**Comment by CarloVarni** [comment]
forgot to ping you @paulgessinger . This is ready for review

File: Examples/Python/python/acts/examples/itk.py [hunk 1]
```diff
@@ -445,9 +445,7 @@ def itkSeedingAlgConfig(
         if highOccupancyConfig == True:
             rMaxGridConfig = 250 * u.mm
             deltaRMax = 200 * u.mm
-            zBinsCustomLooping = [1, 11, 2, 10, 3, 9, 6, 4, 8, 5, 7]
-            # variables that are only used for highOccupancyConfig configuration:
-            skipZMiddleBinSearch = 2
+            zBinsCustomLooping = [2, 10, 3, 9, 6, 4, 8, 5, 7]
```
Comment (paulgessinger) [inline Examples/Python/python/acts/examples/itk.py:448]: Sorry, I'm slow. What does this end up doing? Are these global middle z bins that the iterator visits? Does this not depend on the seed combination?

Comment (CarloVarni) [reply Examples/Python/python/acts/examples/itk.py:448]: This basically allows you to search for the middle space point only in some regions of the detector, as defined by the grid. Also, you can now specify these custom loopings for any axis in the grid ... potentially you can do seeding only in the forward region of the detector by being smart with these arrays. If you add an r-axis to the grid you can also only loop in the innermost part of the detector with no `c++` change

**Review by paulgessinger — APPROVED** [review]

**Pull request merged**